.github/workflows/secret-scanning.yml

```
@@ -0,0 +1,205 @@
+  secret-scanning:
+    name: 🔍 Comprehensive Secret Scanning
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: 📥 Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # Full history for thorough scanning
+        
+    - name: 🐍 Set up Python
+      uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)